CHANGES.rst

```diff
@@ -185,6 +185,19 @@ lastframe
 
 - Removed unnecessary copies, and created a single copy at step.py level. [#8676]
 
+lib
+---
+
+- Created two functions to workaround the memory increase due to
+  datamodel.open. The new functions in the basic_ultis script are
+  ``use_datamodels`` and ``copy_datamodel``. The first function
+  determines if it is necessary to open a datamodel and returns
+  it or simply returns the datamodel given as input. The second
+  function makes a copy of the datamodel depending on whether
+  the step is being run as part of a pipeline or not. Both
+  functions are now called in all steps of the Detector1
+  pipeline. [#8588]
+
```

Comment:
```suggestion
```

now that #8671 is merged (switching change log handling to `towncrier`) this change log entry should be a file in `changes/` instead:
```shell
echo "Created two functions to workaround the memory increase due to ``datamodel.open``. The new functions in the ``basic_utils`` script are ``use_datamodels`` and ``copy_datamodel``. The first function determines if it is necessary to open a datamodel and returns it or simply returns the datamodel given as input. The second function makes a copy of the datamodel depending on whether the step is being run as part of a pipeline or not. Both functions are now called in all steps of the Detector1 pipeline." > changes/8588.general.rst
```

(new PRs will include instructions on how to do this)